internal/core/services/room_manager/status.go

```diff
@@ -63,19 +63,19 @@ var validStatusTransitions = map[game_room.GameRoomStatus]map[game_room.GameRoom
 	game_room.GameStatusTerminating: {},
 }
 
-// SetRoomStatus changes the game room status in the storage. It takes into
-// count the current status.
-func (m *RoomManager) SetRoomStatus(ctx context.Context, gameRoom *game_room.GameRoom, status game_room.GameRoomStatus) error {
-	transitions, ok := validStatusTransitions[gameRoom.Status]
+// setRoomStatus changes the game room status in the storage. It takes into
+// account the current status.
+func (m *RoomManager) setRoomStatus(ctx context.Context, currentGameRoom *game_room.GameRoom, newGameRoom *game_room.GameRoom) error {
```

Comment:
I would avoid receiving the "currentGameRoom" as a parameter once someone can make the mistake of calling it twice with the same instance. This could work better if the method searches the room by itself.

Reply:
solved in a30c31b